test/groupby.jl

```diff
@@ -0,0 +1,11 @@
+using DimensionalData, Test, Dates
+using DimensionalData: Dimensions
+using DimensionalData: LookupArrays
+using Statistics
+const DD = DimensionalData
+
+# Make a demo DimArray
+A = rand(X(1:0.1:2), Y(1:20), Ti(DateTime(2000):Day(10):DateTime(2003)))
+# Group by month and even/odd Y axis values
+groupmeans = groupby(A, Ti=month, Y=isodd)
```

Comment:
I think I should do some comparisons with other libraries, just a few. But, in the meantime the printing, specially here is really too much. 

Maybe for starters, don't print any numbers, and just focus on the output groups and dimensions. Print just that, each output group name and corresponding Dims.

Reply:
Good idea would be good to have more examples.

Sorry its not really related to this PR its just how we print arrays of arrays 😭

We need to get them to squash like Base does